taxonomy/utils.py

```diff
@@ -238,3 +242,83 @@ def get_course_jobs(course_key):
             }
         )
     return data
+
+
+def get_translated_course_description(course_key, course_description):
+    """
+    Return translated course description.
+
+    Create translation for course description if translation object doesn't already exist.
+     OR update translation if course description changed from previous description in translation and
+      return the translated course description.
+
+    Arguments:
+        course_key (str): Key of the course whose description needs to be translated.
+        course_description (str): Full course description of the course which needs to be translated.
+
+    Returns:
+        str: Translated full course description.
+    """
+    translation = Translation.objects.filter(source_model_name='Course',
+                                             source_model_field='full_description',
+                                             source_record_identifier=course_key).first()
+    if translation:
+        if translation.source_text != course_description:
+            result = translate_text(course_key, course_description, AUTO, ENGLISH)
+            if not result:
+                return course_description
+            if result['SourceLanguageCode'] == ENGLISH:
+                translation.translated_text = course_description
+            else:
+                translation.translated_text = result['TranslatedText']
+            LOGGER.info(f'[TAXONOMY] Translate course description updated for key: {course_key}')
+            translation.source_text = course_description
+            translation.source_language = result['SourceLanguageCode']
+            translation.save()
+        return translation.translated_text
+    result = translate_text(course_key, course_description, AUTO, ENGLISH)
+    if not result:
+        return course_description
+    if result['SourceLanguageCode'] == ENGLISH:
+        translated_text = course_description
+    else:
+        translated_text = result['TranslatedText']
+
+    translation = Translation.objects.create(source_model_name='Course',
```

Comment:
same as above, indentation need to be updated according to https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines